Description: Update configure.ac to handle modern build systems
Author: Jason Pereira <jason.pereira@sohonet.com>
Last-Update: 2024-03-18
Forwarded: no

--- a/configure.ac
+++ b/configure.ac
@@ -2,7 +2,7 @@
 # Process this file with autoconf to produce a configure script.

 ## Set the version in both instances below here:
-AC_INIT([Statsite], [0.8.1])
+AC_INIT([statsite], [0.8.1], [armon@armon.me])

 STATSITE_MAJOR_VERSION=0
 STATSITE_MINOR_VERSION=8
@@ -16,6 +16,9 @@
 # place where extra autoconf macros are kept
 AC_CONFIG_MACRO_DIR([m4])

+# Add hardening flags support
+CFLAGS="$CFLAGS -fstack-protector-strong -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security"
+
 # define things like _GNU_SOURCE appropriately
 AC_USE_SYSTEM_EXTENSIONS
